openassessment/xblock/leaderboard_mixin.py

```diff
@@ -72,13 +75,16 @@ def render_leaderboard_complete(self, student_item_dict):
         for score in scores:
             if 'file_key' in score['content']:
                 score['file'] = file_upload_api.get_download_url(score['content']['file_key'])
-            if 'text' in score['content']:
-                score['content'] = score['content']['text']
+            if 'text' in score['content'] or 'parts' in score['content']:
```

Comment:
One comment I'd make, given the need for backwards compatibility, is exploring the idea of a utility module for handling submissions that may or may not be multi-part, etc. Then we won't have to have special case back compat code throughout. 

Just a thought.